src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/repositories/WorkbookRepository.kt

```diff
@@ -40,56 +41,49 @@ class WorkbookRepository(
             title = bookCollection.titleKey,
             sort = bookCollection.sort,
             chapters = constructBookChapters(bookCollection),
-            subtreeResources = resourceRepository.getSubtreeResourceInfo(bookCollection)
+            subtreeResources = db.getSubtreeResourceInfo(bookCollection)
         )
     }
 
     private fun constructBookChapters(bookCollection: Collection): Observable<Chapter> {
-        val collections = collectionRepository
-            .getChildren(bookCollection)
-            .flattenAsObservable { list -> list.sortedBy { it.sort } }
-
-        val chapters = collections
-            .concatMapEager { constructChapter(it).toObservable() }
-
-        return chapters.cache()
+        return Observable.defer {
+            db.getChildren(bookCollection)
+                .flattenAsObservable { it }
+                .concatMapEager { constructChapter(it).toObservable() }
+        }.cache()
```

Comment:
Does defer work with cache?

Reply:
Yes, I believe this is verified by the tests chaptersAreLazyLoad and chaptersAreCached.

Reply:
I was curious because the docs made it sound like it creates a new observable every time it's subscribed, but I wasn't sure that cache stopped that behavior. Observable.fromCallable is also supposed to be lazy and defer execution until subscribe, but made no specific mention of making a new observable every time.

I also didn't see the tests, so I appreciate you mentioning it. They got grouped up into one small "this diff is too large to display by default" thing and I totally missed it. Which definitely made me confused as to the name of the PR ahaha.